nvflare/private/fed/client/client_runner.py

```diff
@@ -239,70 +242,65 @@ def _try_run(self):
                     self.log_info(fl_ctx, "run abort signal received")
                     break
 
-                # reset to default fetch interval
-                task_fetch_interval = self.task_fetch_interval
-                self.log_debug(fl_ctx, "fetching task from server ...")
-                task = self.engine.get_task_assignment(fl_ctx)
-                if not task:
-                    self.log_debug(fl_ctx, "no task received - will try in {} secs".format(task_fetch_interval))
-                    continue
+                task_fetch_interval, _ = self.run_one_task(fl_ctx)
 
-                if task.name == SpecialTaskName.END_RUN:
-                    self.log_info(fl_ctx, "server asked to end the run")
-                    break
+    def run_one_task(self, fl_ctx):
+        # reset to default fetch interval
+        task_fetch_interval = self.task_fetch_interval
+        self.log_debug(fl_ctx, "fetching task from server ...")
+        task = self.engine.get_task_assignment(fl_ctx)
+        task_processed = False
```

Comment:
suggest:
move "task fetch" logic outside of this function,
make function signature `def run_one_task(self, task, fl_ctx)`